src/mono/browser/runtime/pthreads/browser/index.ts

```diff
@@ -168,4 +171,36 @@ export function cancelThreads() {
             worker.postMessage({ cmd: "cancel" });
         }
     }
-}
\ No newline at end of file
+}
+
+export function dumpThreads(): void {
+    if (!WasmEnableThreads) return;
+    mono_log_info("Dumping web worker info as seen by UI thread, it could be stale: ");
+    const emptyInfo = {
+        pthreadId: 0,
+        threadPrefix: "          -    ",
+        threadName: "????",
+        isRunning: false,
+        isAttached: false,
+        isExternalEventLoop: false,
+        reuseCount: 0,
+    };
+    const threadInfos: PThreadInfo[] = [
+        Object.assign({}, emptyInfo, monoThreadInfo), // UI thread
+    ];
+    for (const worker of getRunningWorkers()) {
+        threadInfos.push(Object.assign({}, emptyInfo, worker.info));
+    }
+    for (const worker of getUnusedWorkerPool()) {
+        threadInfos.push(Object.assign({}, emptyInfo, worker.info));
+    }
+    threadInfos.forEach((info, i) => {
+        const idx = (i + "").padStart(2, "0");
+        const isRunning = (info.isRunning + "").padStart(5, " ");
+        const isAttached = (info.isAttached + "").padStart(5, " ");
+        const isEventLoop = (info.isExternalEventLoop + "").padStart(5, " ");
+        const reuseCount = (info.reuseCount + "").padStart(3, " ");
+        // eslint-disable-next-line no-console
+        console.info(`${idx} | ${info.threadPrefix}: isRunning:${isRunning} isAttached:${isAttached} isEventLoop:${isEventLoop} reuseCount:${reuseCount} - ${info.threadName}`);
```

Comment:
Shouldn't we stick to logging functions?

```suggestion
        mono_log_info(`${idx} | ${info.threadPrefix}: isRunning:${isRunning} isAttached:${isAttached} isEventLoop:${isEventLoop} reuseCount:${reuseCount} - ${info.threadName}`);
```